### PR TITLE
added irc disentangle dataset

### PR DESCRIPTION
**Pull request by dhruvjoshi1998**
added irc disentanglement dataset

### CODE REVIEW

**Comment by dhruvjoshi1998** [comment]
@lhoestq sorry, this was the only way I was able to fix the pull request

File: datasets/irc_disentangle/README.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+task_categories:
+- sequence-modeling
+- structure-prediction
+task_ids:
+- coreference-resolution
+- dialogue-modeling
+---
+
+
+---
```
Comment (lhoestq) [inline datasets/irc_disentangle/README.md:27]:
```suggestion
---
annotations_creators:
- expert-generated
language_creators:
- crowdsourced
languages:
- en
licenses:
- cc-by-4.0
multilinguality:
- monolingual
size_categories:
- 100K<n<1M
source_datasets:
- original
task_categories:
- sequence-modeling
- structure-prediction
task_ids:
- coreference-resolution
- dialogue-modeling
---
```
formatting

File: datasets/irc_disentangle/README.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+# Dataset Card for [Dataset Name]
```
Comment (lhoestq) [inline datasets/irc_disentangle/README.md:29]: can you add the name here ?

File: datasets/irc_disentangle/README.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+## Dataset Structure
+
+### Data Instances
+
+[More Information Needed]
+
+### Data Fields
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/irc_disentangle/README.md:82]: can you add a brief description of each data field please ? (raw, ascii, etc.)

File: datasets/irc_disentangle/README.md [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+### Languages
+
+English (en)
+
+## Dataset Structure
+
+### Data Instances
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/irc_disentangle/README.md:78]: can you add an example from the dataset /

You can get one by using `load_dataset` and with `dataset["train"][0]`

**Review by lhoestq — COMMENTED** [review]
Nice thanks !

I left a few minor comments.

Also it looks like the dummy_data.zip files are quite big (300KB each).
Can you try to make them smaller ? (like <20KB each ?)

To do so feel free to take a look inside them and remove all the unnecessary files to only keep 1 or 2 examples. For example for the dummy_data.zip file for the `ubuntu` config you can remove the files for the `channel_two` config and then keep only 1 or 2 raw/ascii/tok/annotation file.

Same for the `channel_two` dummy_data.zip file

**Comment by dhruvjoshi1998** [comment]
@lhoestq Thank you for the feedback. I wondering whether I should be passing an 'id' field in the dictionary since the 'connections' reference the 'id' of the linked messages. This 'id' would just be the same as the id_ that is in the yielded tuple.

File: datasets/irc_disentangle/README.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+task_categories:
+- sequence-modeling
+- structure-prediction
+task_ids:
+- coreference-resolution
+- dialogue-modeling
+---
+
+
+---
```
Comment (lhoestq) [inline datasets/irc_disentangle/README.md:25]:
```suggestion
```

File: datasets/irc_disentangle/README.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+  'connections': [1048, 1054, 1055, 1072, 1073], 
+
+  'date': '2004-12-25', 
+
+  'raw': "[03:57] <Xophe> (also, I'm guessing that this isn't a good place to report minor but annoying bugs... what is?)", 
+
+  'tokenized': "<s> ( also , i 'm guessing that this is n't a good place to report minor but annoying bugs ... what is ?) </s>"
+
+}
```
Comment (lhoestq) [inline datasets/irc_disentangle/README.md:92]:
````suggestion
```
{

  'ascii': "[03:57] <Xophe> (also, I'm guessing that this isn't a good place to report minor but annoying bugs... what is?)", 

  'connections': [1048, 1054, 1055, 1072, 1073], 

  'date': '2004-12-25', 

  'raw': "[03:57] <Xophe> (also, I'm guessing that this isn't a good place to report minor but annoying bugs... what is?)", 

  'tokenized': "<s> ( also , i 'm guessing that this is n't a good place to report minor but annoying bugs ... what is ?) </s>"

}
```
````

File: datasets/irc_disentangle/README.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+  'ascii': "[01:04] <Felicia> Chanel: i don't know off hand sorry", 
+
+  'connections': [49, 53], 
+
+  'raw': "[01:04] <Felicia> Chanel: i don't know off hand sorry", 
+
+  'tokenized': "<s> <user> : i do n't know off hand sorry </s>"
+
+}
```
Comment (lhoestq) [inline datasets/irc_disentangle/README.md:108]:
````suggestion
```
{

  'ascii': "[01:04] <Felicia> Chanel: i don't know off hand sorry", 

  'connections': [49, 53], 

  'raw': "[01:04] <Felicia> Chanel: i don't know off hand sorry", 

  'tokenized': "<s> <user> : i do n't know off hand sorry </s>"

}
```
````

**Comment by lhoestq** [comment]
Yes indeed it would be cool to have the ids in the dictionary. This way the dataset can be shuffled and all without losing information about the connections. Can you add it if you don't mind ?

**Review by lhoestq — COMMENTED** [review]
I added some minor formatting changes

**Comment by lhoestq** [comment]
Thanks :) could you also add the ids in the dictionary since they're useful for the connection links ?

**Comment by lhoestq** [comment]
Thanks !
Also it looks like the dummy_data.zip were regenerated and are now back to being too big (300KB each).
Can you reduce their sizes ? You can actually just revert to the ones you had before the last commit

**Review by lhoestq — APPROVED** [review]
Thanks for adding this one !

Sorry for the delay on reviewing this one
Looks all good to me now

**Pull request merged**